sim/src/ics/component.h

```diff
@@ -0,0 +1,30 @@
+#ifndef BENTOBOX_COMPONENT_H
+#define BENTOBOX_COMPONENT_H
+
+#include <concepts>
+
+namespace ics {
+    template<class T>
+    concept Component = std::semiregular<T>
+            && std::same_as<decltype(T::isActive), bool>;
+
+    // Needed for Index concept
+    // See: https://stackoverflow.com/q/37655113/4428725
+    namespace archetypes {
+        struct Component {
+            bool isActive = true;
+            bool operator==(const Component&) const = default;
+        };
+    }
+
+    static_assert(Component<archetypes::Component>);
+
+    struct DefaultComponent {
+        bool isActive = true;
+        bool operator==(const DefaultComponent&) const = default;
+    };
+
+    static_assert(Component<DefaultComponent>);
```

Comment:
What is the difference between a default component and a normal component?

Reply:
The `archetypes::Component` is used only for concepts. When defining the index concept, I can only enforce that the argument follows the component concept if I have some type that represents the minimal concept.

The DefaultComponent is just a reference implementation of a component. Whether or not you choose to use it is up to you. As long as your component fulfils the concept, it will work. Currently, the only use of `DefaultComponent` is in tests, so that I can create components without worrying about fulfilling the concept.

In summary: `archetypes::Component` is used to represent the concept, `DefaultComponent` is used as a reference implementation. As they serve separate purposes, they should be separate.

Reply:
Do you intend for components to be created inheriting `DefaultComponent`? Is there a use for `DefaultComponent` outside of tests? If its only tests than I propose a move to tests and rename to `TestComponent`

Reply:
I intend that the components that I write in C++ will use DefaultComponent. User-defined components will likely be a class as a lot of user-defined attributes should be hidden and accessed only through methods.